.github/labeler.yml

```diff
@@ -0,0 +1,9 @@
+test:
+- any: ['**/*_test.go']
+
+server:
+- any: ['**/*.go']
+  all: ['!warehouse/**/*']
+
+warehouse:
+- any: ['warehouse/**/*']
```

Comment:
There's many more we could add here but this is a really good starting point to check out how this looks without putting too many labels.